pkg/initialize/run.go

```diff
@@ -60,20 +63,202 @@ func Run(isNonInitCmd bool) error {
+	}
+
+	for _, ruleMetadata := range ruleMetadataList {
+		policy, err := getCommercialPolicy(ruleMetadata)
+		if err != nil {
+			return err
+		}
+
+		err = saveCommercialPolicies(policy, policyRepoPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getCommercialPolicy(ruleMetadata commercialPolicyMetadata) (commercialPolicy, error) {
+	var policy commercialPolicy
+	var templateArgs map[string]interface{}
+
+	policy.regoTemplate = "package accurics\n\n" + ruleMetadata.RuleTemplate
+	policy.metadataFileName = ruleMetadata.RuleReferenceID + ".json"
+	policy.resourceType = ruleMetadata.ResourceType
 
+	policy.policyMetadata.Name = ruleMetadata.RuleName
+	policy.policyMetadata.File = ruleMetadata.RegoName + ".rego"
+	policy.policyMetadata.ResourceType = ruleMetadata.ResourceType
+	policy.policyMetadata.Severity = ruleMetadata.Severity
+	policy.policyMetadata.Description = ruleMetadata.RuleDisplayName
+	policy.policyMetadata.ReferenceID = ruleMetadata.RuleReferenceID
+	policy.policyMetadata.ID = ruleMetadata.RuleReferenceID
+	policy.policyMetadata.Category = ruleMetadata.Category
+	policy.policyMetadata.Version = ruleMetadata.Version
+
+	templateString, ok := ruleMetadata.RuleArgument.(string)
+	if !ok && templateString != "" {
+		return policy, fmt.Errorf("incorrect rule argument type, must be a []byte")
+	}
+	err := json.Unmarshal([]byte(templateString), &templateArgs)
+	if err != nil {
+		return policy, fmt.Errorf("error could not unmarshal rule arguments into map[string]interface{}: '%v'", err)
+	}
+	policy.policyMetadata.TemplateArgs = templateArgs
+
+	return policy, nil
+}
+
+func saveCommercialPolicies(policy commercialPolicy, policyRepoPath string) error {
+	const tabSpace = "    "
+
+	policy.policyMetadata.PolicyType = getCSP(policy.resourceType)
+	cspDir := filepath.Join(policyRepoPath, policy.policyMetadata.PolicyType)
+	err := ensureDir(cspDir)
+	if err != nil {
+		return err
+	}
+
+	resourceDir := filepath.Join(cspDir, policy.resourceType)
+	err = ensureDir(resourceDir)
+	if err != nil {
+		return err
+	}
+
+	var buffer bytes.Buffer
+	encoder := json.NewEncoder(&buffer)
+	encoder.SetEscapeHTML(false)
+	encoder.SetIndent("", tabSpace)
+	err = encoder.Encode(policy.policyMetadata)
+	if err != nil {
+		return fmt.Errorf("error could not marshal json object into byte array: '%v'", err)
+	}
+
+	metadata := buffer.Bytes()
+	metadata = bytes.TrimRight(metadata, "\n")
+	metaDataPath := filepath.Join(resourceDir, policy.metadataFileName)
+	err = ioutil.WriteFile(metaDataPath, metadata, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("error could not write rule metadata file on disk: '%v'", err)
+	}
+
+	regoPath := filepath.Join(resourceDir, policy.policyMetadata.File)
+	err = ioutil.WriteFile(regoPath, []byte(policy.regoTemplate), filePermissionBits)
+	if err != nil {
+		return fmt.Errorf("error could not write rego code file on disk: '%v'", err)
```

Comment:
we can use following error messages
`fmt.Errorf("error occurred while writing rego file, %w", err)`
or
`fmt.Errorf("could not write rego file, error: %w", err)`